### PR TITLE
Add support for attachment upload and update request dependency

### PR DESCRIPTION
**Pull request by adrianfalleiro**
Added support for file attachment. Checks if there is an attachment parameter passed and if so, sets the formData.

### CODE REVIEW

**Comment by adrianfalleiro** [comment]
Hi @adunkman, I wrote some tests. I also had to update to Should.js 5.1.0.
I figured out how to test for a stream from this StackOverflow thread: http://stackoverflow.com/a/28564000

File: package.json [hunk 1]
```diff
@@ -13,11 +13,11 @@
   "dependencies": {
     "coffee-script": "1.3.2",
     "oauth": "0.9.7",
-    "request": "2.12.0"
+    "request": "2.51.0"
   },
   "devDependencies": {
     "mocha": "1.1.0",
-    "should": "0.6.3"
+    "should": "5.1.0"
```
Comment (adunkman) [inline package.json]: Haha, that’s quite an upgrade! :p

**Comment by adrianfalleiro** [comment]
Yeah, its quite a bump, I had issues with the instanceOf Should.js method, so I had to bump it. But I also rewrote the tests that were failing because of Should.js API changes so now it passes the test.

**Comment by adunkman** [comment]
Nice! It seems to work great on my machine, but trying to figure out why the Travis build is failing.

**Comment by adunkman** [comment]
Seems to be an issue with `assert-plus` and versions of node < 0.10, although it’s stated to support 0.8 and up.

I’m just going to drop support/test runs for 0.8 and see if anybody notices. It’s about time people on 0.8 upgraded, anyways! :)

**Comment by adrianfalleiro** [comment]
Cheers, thanks for merging :)

**Comment by adunkman** [comment]
Published as `v1.1.0`. Thanks for the PR!

**Pull request merged**